src/components/visit/visitInstance.tsx

```diff
@@ -0,0 +1,241 @@
+import React from 'react'
+import ChevronDownIcon from '@heroicons/react/outline/ChevronDownIcon'
+import { updateUserData, User, Visit } from 'databaseIntigration'
+
+import EditButton from '@/components/visit/Buttons'
+
+type VisitInstanceProps = Visit
+
+interface VisitInstanceState extends User {
+  isEditable: boolean
+  isOpen: boolean
+  isLoaded: boolean
+}
+
+class VisitInstance extends React.Component<
```

Comment:
I would avoid using class components unless you absolutely have to. There are only a few rare cases like error boundaries where they are still necessary.